CatAnalyzer/test/create-batch

```diff
@@ -0,0 +1,473 @@
+done""" % (' '.join(self.outFileNames), self.transferCmd, self.dest)
+        print>>fout, "echo FINISHED `date` # >> %s/submit.log" % self.jobDir
+        fout = None
+        os.chmod(runFileName, 0755)
+
+        ## Write submit script
+        fout = open("%s/submit.log" % self.jobDir, "w")
+        print>>fout, "SUBMIT STARTED", self.jobName
+        fout = None
+        os.chmod("%s/submit.log" % self.jobDir, 0666)
+
+        print "@@ Writing submit script..."
+        submitFileName = "%s/submit.sh" % self.jobDir
+        fout = open(submitFileName, "w")
+        print>>fout, "#!/bin/bash"
+        print>>fout, self.submitCmd
+        fout = None
+        os.chmod(submitFileName, 0755)
+
+        # Condor jobs
+        jdlOut = open("%s/submit.jds" % self.jobDir, "w")
+        print>>jdlOut, "# Job description file for condor job", self.jobName
+        print>>jdlOut, """executable = run.sh
+universe   = vanilla
+arguments  = $(Process)
+log = condor.log
+
+getenv     = True
+should_transfer_files = YES
+when_to_transfer_output = ON_EXIT
+output = job_$(Process).log
+error = job_$(Process).err
+transfer_input_files = job.tgz"""
+        if self.transferCmd == '':
+            print>>jdlOut, "transfer_output_files = ", (",".join([os.path.join(self.jobBase, x) for x in self.outFileNames]))
+            remapStrs = ["{0}.{1}={0}_$(Process).{1}".format('.'.join(x.split('.')[:-1]), x.split('.')[-1]) for x in self.outFileNames]
+            print>>jdlOut, 'transfer_output_remaps = "%s"' % (';'.join(remapStrs))
+        print>>jdlOut, "queue",  self.nSection
+        jdlOut = None
+
+    def archive(self):
+        ## Archive libraries and other stuffs
+        archiveCmd = "tar czf {0}/job.tgz {1} --xform='s:{1}:{2}:' -P".format(self.jobDir, self.cmsswBase, os.path.basename(self.cmsswBase))
+        archiveCmd += " --exclude tmp --exclude '*.root' --exclude 'job.tgz' --exclude .git --exclude-tag-all=.create-batch"
+        print "@@ Archive files for job submission..." 
+        os.system(archiveCmd)
+        os.system("rm -f %s/job_*_cfg.py" % self.jobDir)
+
+        tmpFile = open("%s/.create-batch" % self.jobDir, "w")
+        tmpFile.close()
+
+    def submit(self):
+        if self.doSubmit:
+            print "@@ Submitting jobs..."
+            if self.site == "UOS":
+                os.system("ssh uosaf0008 'cd %s;./submit.sh'" % self.jobDir)
+            elif self.site == "CERN" and not os.environ['HOSTNAME'].startswith("lxplus"):
+                os.system("ssh lxplus.cern.ch 'cd %s;./submit.sh'" % self.jobDir)
+            else:
+                os.system("cd %s;./submit.sh" % self.jobDir)
+        else:
+            print "@@ Jobs are prepared. You can submit jobs with following command:"
+            print "cd %s;./submit.sh" % (self.jobDir)
+
+## Create job configure
+jobConfig = TheJobConfig(opts)
+jobConfig.initialiseWorkspace()
+jobConfig.archive()
+jobConfig.submit()
+
+print "@@ Done."
```

Comment:
What is different to the central create-batch script?
I'd like to merge this if there's any improvement.